eurostag-converter/src/main/java/com/powsybl/eurostag/converter/EurostagEchExport.java

```diff
@@ -562,9 +566,315 @@ private void createTransformers(EsgNetwork esgNetwork, EsgGeneralParameters para
+                if (regulatingBus.getId() != null) {
+                    regulatingMode = EsgThreeWindingTransformer.RegulatingMode.VOLTAGE;
+                    zbusr = new Esg8charName(dictionary.getEsgId(regulatingBus.getId()));
+                    voltr = rtcL3.getTargetV();
+                    ktap8 = rtcL3.getTapPosition() - rtcL3.getLowTapPosition() + 1;
+                    ktpnom = rtcL3.getStepCount() / 2 + 1;
+                }
+            }
+
+            double[] ucc12 = new double[nbTaps];
+            double[] ucc13 = new double[nbTaps];
+            double[] ucc23 = new double[nbTaps];
+
+            double[] uno1 = new double[nbTaps];
+            double[] uno2 = new double[nbTaps];
+            double[] uno3 = new double[nbTaps];
+
+            for (int i = 0; i < nbTaps; i++) {
+
+                uno1[i] = nomiU1;
+                uno2[i] = nomiU2;
+                uno3[i] = nomiU3;
+
+                //initialization with the current tap
+                if (ptcL1 != null) {
+                    uno1[i] = uno1[i] / ptcL1.getCurrentStep().getRho();
+                }
+                if (ptcL2 != null) {
+                    uno2[i] = uno2[i] / ptcL2.getCurrentStep().getRho();
+                }
+                if (ptcL3 != null) {
+                    uno3[i] = uno3[i] / ptcL3.getCurrentStep().getRho();
+                }
+
+                double uccT1 = 100 * Math.hypot(r1pu, x1pu);
+                if (x1pu < 0) {
+                    uccT1 = x1pu * 100;
+                }
+                double uccT2 = 100 * Math.hypot(r2pu, x2pu);
+                if (x2pu < 0) {
+                    uccT2 = x2pu * 100;
+                }
+                double uccT3 = 100 * Math.hypot(r3eqpu, x3eqpu);
+                if (x3eqpu < 0) {
+                    uccT3 = x3eqpu * 100;
+                }
+                if (rtci == 1) {
+                    double r1putap = getValue(r1pu, rtcL1.getStep(rtcL1.getLowTapPosition() + i).getR(), 0);
+                    double x1putap = getValue(x1pu, rtcL1.getStep(rtcL1.getLowTapPosition() + i).getX(), 0);
+                    uccT1 = 100 * Math.hypot(r1putap, x1putap);
+                    uno1[i] = uno1[i] / rtcL1.getStep(rtcL1.getLowTapPosition() + i).getRho();
+                    if (rtcL2 != null) {
+                        uno2[i] = uno2[i] / rtcL2.getCurrentStep().getRho();
+                    }
+                    if (rtcL3 != null) {
+                        uno3[i] = uno3[i] / rtcL3.getCurrentStep().getRho();
+                    }
+                } else if (rtci == 2) {
+                    double r2putap = getValue(r2pu, rtcL2.getStep(rtcL2.getLowTapPosition() + i).getR(), 0);
+                    double x2putap = getValue(x2pu, rtcL2.getStep(rtcL2.getLowTapPosition() + i).getX(), 0);
+                    uccT2 = 100 * Math.hypot(r2putap, x2putap);
+                    if (rtcL1 != null) {
```

Comment:
You don't have a line to change uno2 here, it is normal?

Reply:
no, you are right, it is missing